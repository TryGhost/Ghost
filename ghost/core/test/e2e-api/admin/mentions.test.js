const {agentProvider, fixtureManager, mockManager, matchers} = require('../../utils/e2e-framework');
const {anyObjectId, anyISODateTime, anyString} = matchers;

const matchMentionShallowIncludes = {
    id: anyObjectId,
    source: anyString,
    target: anyString,
    timestamp: anyISODateTime,
    source_title: anyString
};

describe('Mentions API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        // TODO: test various users' access
        await fixtureManager.init('users','mentions');
        await agent.loginAsOwner();
    });

    afterEach(function () {
        mockManager.restore();
    });

    it('Can browse with limits', async function () {
        await agent.get('mentions/?limit=2')
            .expectStatus(200)
            .matchBodySnapshot({
                mentions: new Array(2).fill(matchMentionShallowIncludes)
            });
    });
});
