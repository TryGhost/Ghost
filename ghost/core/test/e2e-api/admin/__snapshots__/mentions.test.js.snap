// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mentions API Can browse with limits 1: [body] 1`] = `
Object {
  "mentions": Array [
    Object {
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "payload": null,
      "resource": null,
      "source": Any<String>,
      "source_author": null,
      "source_excerpt": null,
      "source_favicon": null,
      "source_featured_image": null,
      "source_site_title": "source.com",
      "source_title": Any<String>,
      "target": Any<String>,
      "timestamp": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "verified": false,
    },
    Object {
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "payload": null,
      "resource": null,
      "source": Any<String>,
      "source_author": null,
      "source_excerpt": null,
      "source_favicon": null,
      "source_featured_image": null,
      "source_site_title": "anothersource.com",
      "source_title": Any<String>,
      "target": Any<String>,
      "timestamp": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "verified": false,
    },
  ],
  "meta": Object {
    "pagination": Object {
      "limit": 2,
      "next": 2,
      "page": 1,
      "pages": 2,
      "prev": null,
      "total": 3,
    },
  },
}
`;
